File: src/main/java/com/tms/csp/transforms/OneArgOr.java
package com.tms.csp.transforms;

import com.tms.csp.ast.Exp;

/**
 * Replaces each Or(a) with a
 */
public class OneArgOr extends BaseTransformer {

    @Override
    protected Exp local(Exp in) {
        return in.getArg();
    }

    @Override
    protected boolean executeLocal(Exp in) {
        return in.isOr() && in.argCount() == 1;
    }

}

File: src/main/java/com/tms/csp/transforms/OneArgOr.kt
package com.tms.csp.transforms

import com.tms.csp.ast.Exp

/**
 * Replaces each Or(a) with a
 */
class OneArgOr : BaseTransformer() {

    override fun local(inExp: Exp): Exp? {
        return inExp.arg
    }

    override fun executeLocal(inExp: Exp): Boolean {
        return inExp.isOr && inExp.argCount() == 1
    }

}
